book/src/implemented-proposals/durable-tx-nonces.md

```diff
@@ -66,21 +63,43 @@ WithdrawInstruction(to, lamports)
   success
 ```
 
-A client wishing to use this feature starts by creating a nonce account and
-depositing sufficient lamports as to make it rent-exempt. The resultant account
-will be in the `Uninitialized` state with no stored hash and thus unusable.
-
-The `Nonce` instruction is used to request that a new nonce be stored for the
-calling account. The first `Nonce` instruction run on a newly created account
-will drive the account's state to `Initialized`. As such, a `Nonce` instruction
-MUST be issued before the account can be used.
-
-To discard a `NonceAccount`, the client should issue a `Withdraw` instruction
-which withdraws all lamports, leaving a zero balance and making the account
-eligible for deletion.
-
-`Nonce` and `Withdraw` instructions each will only succeed if the stored
-blockhash is no longer resident in sysvar.recent_blockhashes.
+A client wishing to use this feature starts by creating a nonce account under 
+the system program. This account will be in the `Uninitialized` state with no
+stored hash and thus unusable.
+
+To initialize a newly created account, a `NonceInitialize` instruction must be
+issued. This instruction takes one parameter, the `Pubkey` of the account's
+[ authority](../offline-signing/durable-nonce.md#nonce-authority). Nonce accounts
+must be [rent-exempt](rent.md#two-tiered-rent-regime) to meet the data-persistence
+requirements of the feature and as such, require that sufficient lamports be
+deposited into the account before it can be initialized. Upon successful
+initialization, the cluster's most recent blockhash is stored along with specified
+nonce authority `Pubkey`.
+
+The `NonceAdvance` instruction is used to manage the account's stored nonce
+value. It stores the cluster's most recent blockhash with the account's state
+data, failing if that matches the value already stored there. This check prevents
+replaying transactions within the same block.
+
+Due to nonce accounts' [rent-exempt](rent.md#two-tiered-rent-regime) requirement,
+a custom withdraw instruction is used to move funds out of the account.
+The `NonceWithdraw` instruction takes a single argument, lamports to withdraw,
+and enforces rent-exemption by preventing the account's balance from falling
+below the rent-exempt minimum. An exception to this check is if the final balance
+would be zero lamports, which makes the account eligible for deletion. This
+account closure detail has an additional requirement that the stored nonce
+value be advanceable as per `NonceAdvance`.
```

Comment:
I actually don't understand what you're saying here (and I haven't looked at the implementation to check). Are you saying that if a withdraw transaction will close an account, the nonce will advance first, and then the account will be deleted?

Reply:
We won't advance the nonce since the account will be deleted anyway, but the same conditions must hold true for a withdraw to zero the balance

Reply:
Oh, I think I see. So it will only make the withdrawal and delete the account if the cluster's most recent blockhash is different than the currently stored value. Are there more criteria?
Maybe rewrite this last sentence as: "This complete withdrawal can only succeed if the stored nonce is advanceable (ie. the cluster's most recent blockhash is different than the currently stored value)."

Reply:
Yep!  I'll clarify